pkg/util/slice/slice.go

```diff
@@ -68,3 +68,17 @@ func ContainsString(slice []string, s string, modifier func(s string) string) bo
 	}
 	return false
 }
+
+// removeDuplicates remove the duplicate value in the slice and keep the order unchanged
+func RemoveDuplicates(s []string) (ret []string) {
+	encountered := map[string]bool{}
+	for i := range s {
+		if encountered[s[i]] {
+			continue
+		} else {
+			encountered[s[i]] = true
+			ret = append(ret, s[i])
```

Comment:
can we pre-allocate ret?

Reply:
We can't, because we can not know the length of return.